src/libcore/intrinsics.rs

```diff
@@ -940,6 +940,7 @@ extern "rust-intrinsic" {
     /// }
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_unstable(feature = "const_transmute")]
     pub fn transmute<T, U>(e: T) -> U;
```

Comment:
So we don't use `const fn` to mark intrinsics as const-evaluable, but we do use an attribute here to mark its const-nature as unstable? That is somewhat inconsistent I have to say.

Reply:
Enabling the use of `const fn` for intrinsics is orthogonal and definitely on my todo list

Reply:
That's fine, I am just saying that this "intermediate state" is somewhat unfortunate. But not a blocker.

This made me realize though that really we should have `#[const_stable(since)]` as well; for stable `const fn` right now there is no indication in the docs since when they are stable as `const fn`... but that's orthogonal to this PR, obviously.

Reply:
I will tackle this now in a separate PR